File: BACKEND-SistemaGestaoEstoqueCaixa/src/main/java/com/example/SistemaRayssaWagner/SistemaGestaoEstoqueCaixa/dto/VendaResponse.java
package com.example.SistemaRayssaWagner.SistemaGestaoEstoqueCaixa.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

public record VendaResponse(
        Long id,
        String usuarioNome,
        BigDecimal valorTotal,
        BigDecimal valorRecebido,
        BigDecimal troco,
        LocalDateTime dataHora,
        List<ItemVendaResponse> itens
) {

}

File: FRONTEND-SistemaGestaoEstoqueCaixa/sistema-estoque-front/src/app/models/vendaResponse.ts
import { ItemVendaResponse } from "./itemVendaResponse";

export interface VendaResponse {
  id: number;
  usuarioNome: string;
  valorTotal: number;
  valorRecebido: number;
  troco: number;
  dataHora: string;
  itens: ItemVendaResponse[];
}